R/applyChangeMeta.R

```diff
@@ -187,29 +187,33 @@ recode_labels <- function(labels, changeTable, existingMeta) {
     single_simpleChanges <- simpleChanges[simpleChanges$varName == var_name, ]
 
     ## Deal with meta data conflicts:
-    # if values will be recoded into each other, do not raise meta data conflict!
-    existing_value_vec <- !is.na(single_simpleChanges$value_new) & single_simpleChanges$value_new %in% single_labels$value
-    values_to_be_recoded <- labels[single_simpleChanges[, "row_num_in_labels"], "value"]
-    existing_value_vec <- existing_value_vec & !single_simpleChanges$value_new %in% values_to_be_recoded
+    values_being_recoded <- labels[single_simpleChanges[, "row_num_in_labels"], "value"]

```

Comment:
Nit: I would prefer `single_simpleChanges$row_num_in_labels` over `single_simpleChanges[, "row_num_in_labels"]`. Or is there any advantage using the latter version?

Reply:
Mh, I think there are certain advantages and disadvantes to all three of `$`, `[` and `[[`. To me, the biggest advantage of `[` is

``` r
mtcars$test
#> NULL
mtcars[["test"]]
#> NULL
mtcars[, "test"]
#> Error in `[.data.frame`(mtcars, , "test"): undefined columns selected
```
<sup>Created on 2023-10-05 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

However, to me the biggest disadvantage is the reliance on `drop = TRUE`, both from a logical point as well as practical point regarding compatability with `tibble` and `data.table`:

``` r
library(tibble)
library(data.table)

mtcars_tibble <- as_tibble(mtcars)
mtcars_dt <- as.data.table(mtcars)

mtcars_tibble$cyl
#>  [1] 6 6 4 6 8 6 8 4 4 6 6 8 8 8 8 8 8 4 4 4 4 8 8 8 8 4 4 4 8 6 8 4
mtcars_tibble[, "cyl"]
#> # A tibble: 32 × 1
#>      cyl
#>    <dbl>
#>  1     6
#>  2     6
#>  3     4
#>  4     6
#>  5     8
#>  6     6
#>  7     8
#>  8     4
#>  9     4
#> 10     6
#> # ℹ 22 more rows
mtcars_tibble[["cyl"]]
#>  [1] 6 6 4 6 8 6 8 4 4 6 6 8 8 8 8 8 8 4 4 4 4 8 8 8 8 4 4 4 8 6 8 4

mtcars_dt$cyl
#>  [1] 6 6 4 6 8 6 8 4 4 6 6 8 8 8 8 8 8 4 4 4 4 8 8 8 8 4 4 4 8 6 8 4
mtcars_dt[, "cyl"]
#>     cyl
#>  1:   6
#>  2:   6
#>  3:   4
#>  4:   6
#>  5:   8
#>  6:   6
#>  7:   8
#>  8:   4
#>  9:   4
#> 10:   6
#> 11:   6
#> 12:   8
#> 13:   8
#> 14:   8
#> 15:   8
#> 16:   8
#> 17:   8
#> 18:   4
#> 19:   4
#> 20:   4
#> 21:   4
#> 22:   8
#> 23:   8
#> 24:   8
#> 25:   8
#> 26:   4
#> 27:   4
#> 28:   4
#> 29:   8
#> 30:   6
#> 31:   8
#> 32:   4
#>     cyl
mtcars_dt[["cyl"]]
#>  [1] 6 6 4 6 8 6 8 4 4 6 6 8 8 8 8 8 8 4 4 4 4 8 8 8 8 4 4 4 8 6 8 4
```
<sup>Created on 2023-10-05 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

I think this again could be something for the code convention group to discuss 🤷 ?

Reply:
(I should mention that I am aware that I am very inconsistent throughout the package regarding the use of `[` and `$`. This is something that could be optimized but has a rather low priority from my perspective.)

Reply:
This is a good point. I agree that consistency across packages is a big plus!

We could also discuss this on our code convention session (or at least open the discussion). These arguments are already valuable as a starting point 💪